### PR TITLE
chore: Adds go vet analyzers in CI

### PR DESCRIPTION
**Pull request by patriciareinoso**
Enable all go vet analyzers in CI
Fixes `fieldalignment` and `nilness` analyzer checks

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: context/context.go [hunk 1]
```diff
@@ -55,10 +55,10 @@ type NSSFContext struct {
 	RegisterIPv4 string
 	// HttpIpv6Address string
```
Comment (gab-arrobo) [inline context/context.go:56]: What about deleting this line? If so, it can be done in another PR. What do you think?
```suggestion
```

File: producer/subscription_id_document.go [hunk 2]
```diff
@@ -32,12 +31,6 @@ func HandleNSSAIAvailabilityUnsubscribe(request *http_wrapper.Request) *http_wra
...
 		return http_wrapper.NewResponse(http.StatusNoContent, nil, nil)
-	} else if problemDetails != nil {
-		return http_wrapper.NewResponse(int(problemDetails.Status), nil, problemDetails)
 	}
-	problemDetails = &models.ProblemDetails{
-		Status: http.StatusForbidden,
-		Cause:  "UNSPECIFIED",
-	}
-	return http_wrapper.NewResponse(http.StatusForbidden, nil, problemDetails)
+	return http_wrapper.NewResponse(int(problemDetails.Status), nil, problemDetails)
```
Comment (gab-arrobo) [inline producer/subscription_id_document.go:35]: Why these changes? Given that `problemDetails` is a struct (`ProblemDetails`), should not the comparison be against an `empty` ProblemDetails struct?

**Review by gab-arrobo — COMMENTED** [review]
Have you run an end-to-end test using AiaB, OnRamp or other system to make sure that the proposed changes (order of the fields, etc.) do not break anything?

File: producer/subscription_id_document.go (hunk 2 above)
Comment (patriciareinoso) [reply producer/subscription_id_document.go]: the nilness go vet analyzer sent an error on line 35 saying that checking against nil was not necessary, it could just be an else. 
Once line 35 becomes an else, it's more clear that lines 38-42 are unreachable.
I added a check against an empty `ProblemDetails` can you check?

**Review by gab-arrobo — APPROVED** [review]
+1 (tested changes with AiaB)
